app/scripts/lib/nonce-tracker.js

```diff
@@ -83,6 +76,43 @@ class NonceTracker {
     return mutex
   }
 
+  async _getNetworkNonceAndDetails (address) {
+    // calculate next nonce
+    // we need to make sure our base count
+    // and pending count are from the same block
+    const currentBlock = await this._getCurrentBlock()
+    const blockNumber = currentBlock.blockNumber
+    const pendingTransactions = this.getPendingTransactions(address)
```

Comment:
Both methods (`getNetworkNonceAndDetails`, `getLocalNextNonce`) call `getPendingTransactions`. This is probably not going to cause any bugs, it's just redundant.